tests/countries/test_anguilla.py

```diff
@@ -0,0 +1,337 @@
+        )
+        self.assertHolidayName(name, range(2001, 2050))
+
+    def test_august_thursday_day(self):
+        name = "August Thursday"
+        self.assertHolidayName(
+            name,
+            "2011-08-04",
+            "2012-08-09",
+            "2013-08-08",
+            "2014-08-07",
+            "2015-08-06",
+            "2016-08-04",
+            "2017-08-10",
+            "2018-08-09",
+            "2019-08-08",
+            "2020-08-06",
+            "2021-08-05",
+            "2022-08-04",
+            "2023-08-10",
+            "2024-08-08",
+        )
+        self.assertHolidayName(name, range(2001, 2050))
+
+    def test_constitution_day(self):
+        name = "Constitution Day"
+        self.assertHolidayName(
+            name,
+            "2011-08-05",
+            "2012-08-10",
+            "2013-08-09",
+            "2014-08-08",
+            "2015-08-07",
+            "2016-08-05",
+            "2017-08-11",
+            "2018-08-10",
+            "2019-08-09",
+            "2020-08-07",
+            "2021-08-06",
+            "2022-08-05",
+            "2023-08-11",
+            "2024-08-09",
+        )
+        self.assertHolidayName(name, range(2001, 2050))
+
+    def test_national_heroes_and_heroines_day(self):
+        name = "Separation Day"
+        self.assertHolidayName(name, (f"{year}-12-19" for year in range(2001, 2011)))
+        dt = ("2004-12-17", "2009-12-18", "2010-12-17")
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertNoNonObservedHoliday(dt)
+
+        name = "National Heroes and Heroines Day"
+        self.assertHolidayName(name, (f"{year}-12-19" for year in range(2011, 2050)))
+        dt = ("2015-12-18", "2020-12-18", "2021-12-17")
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_christmas_day(self):
+        name = "Christmas Day"
+        self.assertHolidayName(name, (f"{year}-12-25" for year in range(2001, 2050)))
+        dt = (
+            "2004-12-27",
+            "2005-12-27",
+            "2010-12-27",
+            "2011-12-27",
+            "2016-12-27",
+            "2021-12-27",
+            "2022-12-27",
+        )
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_boxing_day(self):
+        name = "Boxing Day"
+        self.assertHolidayName(name, (f"{year}-12-26" for year in range(2001, 2050)))
+        dt = ("2015-12-28", "2020-12-28")
+        self.assertHolidayName(f"{name} (observed)", dt)
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_special_holiday(self):
+        name = "Royal Wedding of Prince William & Kate Middleton"
+        self.assertHolidayName(name, "2011-04-29")
+
+        name = "Diamond Jubilee Celebration of Her Majesty The Queen"
+        self.assertHolidayName(name, "2012-06-04")
+
+        name = "Special Public Holiday"
+        self.assertHolidayName(name, "2025-02-28")
```

Comment:
Let's move this up, there's no need for duplicate `test_special_holiday` test cases
```suggestion
```